src/compute.rs

```diff
@@ -8,54 +9,65 @@ use crate::{
         point::{Compressed, Error as PointError, Point, G},
         scalar::Scalar,
     },
-    util::hash_to_scalar,
+    errors::AggregatorError,
+    util::{expand_to_scalar, hash_to_scalar},
 };
 
 #[allow(non_snake_case)]
-/// Compute a binding value from the party ID, public nonces, and signed message
-pub fn binding(id: &Scalar, B: &[PublicNonce], msg: &[u8]) -> Scalar {
-    let mut hasher = Sha256::new();
-    let prefix = "WSTS/binding";
+/// Compute a binding value from the party ID, public nonces, and signed message using XMD-based expansion.
+pub fn binding(id: &Scalar, B: &[PublicNonce], msg: &[u8]) -> Result<Scalar, EllipticCurveError> {
+    let prefix = b"WSTS/binding";
+
+    // Serialize all input into a buffer
+    let mut buf = Vec::new();
+    buf.extend_from_slice(&id.to_bytes());
 
-    hasher.update(prefix.as_bytes());
-    hasher.update(id.to_bytes());
     for b in B {
-        hasher.update(b.D.compress().as_bytes());
-        hasher.update(b.E.compress().as_bytes());
+        buf.extend_from_slice(b.D.compress().as_bytes());
+        buf.extend_from_slice(b.E.compress().as_bytes());
     }
-    hasher.update(msg);
 
-    hash_to_scalar(&mut hasher)
+    buf.extend_from_slice(msg);
+
+    expand_to_scalar(&buf, prefix)
 }
 
 #[allow(non_snake_case)]
-/// Compute a binding value from the party ID, public nonces, and signed message
-pub fn binding_compressed(id: &Scalar, B: &[(Compressed, Compressed)], msg: &[u8]) -> Scalar {
-    let mut hasher = Sha256::new();
-    let prefix = "WSTS/binding";
+/// Compute a binding value from the party ID, public nonces, and signed message using XMD-based expansion.
+pub fn binding_compressed(
+    id: &Scalar,
+    B: &[(Compressed, Compressed)],
+    msg: &[u8],
+) -> Result<Scalar, EllipticCurveError> {
+    let prefix = b"WSTS/binding";
+
+    // Serialize all input into a buffer
+    let mut buf = Vec::new();
+    buf.extend_from_slice(&id.to_bytes());
 
-    hasher.update(prefix.as_bytes());
-    hasher.update(id.to_bytes());
     for (D, E) in B {
-        hasher.update(D.as_bytes());
-        hasher.update(E.as_bytes());
+        buf.extend_from_slice(D.as_bytes());
+        buf.extend_from_slice(E.as_bytes());
     }
-    hasher.update(msg);
 
-    hash_to_scalar(&mut hasher)
+    buf.extend_from_slice(msg);
+
+    expand_to_scalar(&buf, prefix)
```

Comment:
Yeah, I'm inclined to `.expect` here with an added comment for why this is guaranteed to succeed.